src/Elastic.Apm/Logging/GlobalLogConfiguration.cs

```diff
@@ -0,0 +1,198 @@
+// Licensed to Elasticsearch B.V under one or more agreements.
+// Elasticsearch B.V licenses this file to you under the Apache 2.0 License.
+// See the LICENSE file in the project root for more information
+
+using System;
+using System.Collections;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+
+#if PROFILER_MANAGED_LOADER
+using static Elastic.Apm.Profiler.Managed.Loader.LogEnvironmentVariables;
+namespace Elastic.Apm.Profiler.Managed.Loader;
+#elif PROFILER_MANAGED
+using static Elastic.Apm.Profiler.Managed.LogEnvironmentVariables;
+namespace Elastic.Apm.Profiler.Managed;
+#elif STARTUP_HOOKS
+using static ElasticApmStartupHook.LogEnvironmentVariables;
+namespace ElasticApmStartupHook;
+#else
+using static Elastic.Apm.Logging.LogEnvironmentVariables;
+
+namespace Elastic.Apm.Logging;
+#endif
+
+internal class EnvironmentLoggingConfiguration(IDictionary environmentVariables = null)
+{
+	public IDictionary EnvironmentVariables { get; } = environmentVariables ?? Environment.GetEnvironmentVariables();
+
+	public string GetSafeEnvironmentVariable(string key)
+	{
+		var value = EnvironmentVariables.Contains(key) ? EnvironmentVariables[key]?.ToString() : null;
+		return value ?? string.Empty;
+	}
+
+	public LogLevel? GetLogLevel(params string[] keys)
+	{
+		var level = keys
+			.Select(k => GetSafeEnvironmentVariable(k))
+			.Select<string, LogLevel?>(v => v.ToLowerInvariant() switch
+			{
+				"trace" => LogLevel.Trace,
+				"debug" => LogLevel.Debug,
+				"info" => LogLevel.Information,
+				"warn" => LogLevel.Warning,
+				"error" => LogLevel.Error,
+				"none" => LogLevel.None,
+				_ => null
+			})
+			.FirstOrDefault(l => l != null);
+		return level;
+	}
+
+	public string GetLogDirectory(params string[] keys)
+	{
+		var path = keys
+			.Select(k => GetSafeEnvironmentVariable(k))
+			.FirstOrDefault(p => !string.IsNullOrEmpty(p));
+
+		return path;
+	}
+
+	public bool AnyConfigured(params string[] keys) =>
+		keys
+			.Select(k => GetSafeEnvironmentVariable(k))
+			.Any(p => !string.IsNullOrEmpty(p));
+
+	public GlobalLogTarget? ParseLogTargets(params string[] keys)
+	{
+		var targets = keys
+			.Select(k => GetSafeEnvironmentVariable(k))
+			.FirstOrDefault(p => !string.IsNullOrEmpty(p));
+		if (string.IsNullOrWhiteSpace(targets))
+			return null;
+
+		var logTargets = GlobalLogTarget.None;
+		var found = false;
+
+		foreach (var target in targets.Split(new[] { ';' }, StringSplitOptions.RemoveEmptyEntries))
+		{
+			if (IsSet(target, "stdout"))
+				logTargets |= GlobalLogTarget.StdOut;
+			else if (IsSet(target, "file"))
+				logTargets |= GlobalLogTarget.File;
+			else if (IsSet(target, "none"))
+				logTargets |= GlobalLogTarget.None;
+		}
+		return !found ? null : logTargets;
+
+		bool IsSet(string k, string v)
+		{
+			var b = k.Trim().Equals(v, StringComparison.InvariantCultureIgnoreCase);
+			if (b)
+				found = true;
+			return b;
+		}
+	}
+
+	internal static string GetDefaultLogDirectory() =>
```

Comment:
This is subtly different to the default directory for the otel distro. Is that expected?

Reply:
It's known, since the Otel distro was reworked in its PR for global file logging.

I was going to create a follow up PR to align this.

Reply:
The reason for the follow up PR is that `Elastic.Apm.Profiler.Managed.Loader.csproj` currently is dependency free and `System.Runtime.InteropServices.RuntimeInformation` is not available without a nuget reference. 

Want to tackle that in isolation.